src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClient.cs

```diff
@@ -2,71 +2,70 @@
 
 using System;
 using System.Collections.Generic;
+using System.ComponentModel.Composition;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
-using Microsoft.VisualStudio.LanguageServer.Client;
-using Microsoft.VisualStudio.Threading;
+using Microsoft.CodeAnalysis.Editor;
 using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.ServiceHub.Client;
+using Microsoft.VisualStudio.LanguageServer.Client;
 using Microsoft.VisualStudio.LanguageServices.Remote;
-using Microsoft.CodeAnalysis.Host;
-using System.Composition;
-using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal abstract class AbstractLanguageServerClient : ILanguageClient
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [ContentType(ContentTypeNames.VisualBasicContentType)]
+    [ContentType(ContentTypeNames.CSharpLspContentTypeName)]
+    [ContentType(ContentTypeNames.VisualBasicLspContentTypeName)]
+    [Export(typeof(ILanguageClient))]
+    [ExportMetadata("Capabilities", "WorkspaceStreamingSymbolProvider")]
```

Comment:
this was needed so that VS LSP client would initialize this component when user initiates code search without opening any C# or VB files. At the time there was no other way to initialize LSP server client in this scenario so I introduced this metadata. It would still load this LSP client in non-Roslyn solutions though (but same in GoTo).
Now in 16.3 there is a new extensibility API, which your in-proc component can call to initialize your LSP client at appropriate time, e.g. on Roslyn package load or solution open.
I still prefer "Capabilities" metadata approach though because it avoids requirement to have an in-proc bootstrapper compoennt and also won't load LSP dlls on solution open (which you might have to do in the future as you migrate more features to LSP though).

Reply:
alright. I will keep it for now.